rust/otap-dataflow/crates/engine/src/runtime_pipeline.rs

```diff
@@ -165,29 +163,98 @@ impl<PData: 'static + Debug> RuntimePipeline<PData> {
 
     /// Gets a reference to any node by its ID as a Node trait object
     #[must_use]
-    pub fn get_node(&self, node_id: &NodeId) -> Option<&dyn Node> {
-        match self.nodes.get(node_id)? {
-            NodeType::Receiver => self.receivers.get(node_id).map(|r| r as &dyn Node),
-            NodeType::Processor => self.processors.get(node_id).map(|p| p as &dyn Node),
-            NodeType::Exporter => self.exporters.get(node_id).map(|e| e as &dyn Node),
+    pub fn get_node(&self, node: NodeIndex) -> Option<&dyn Node> {
+        let ndef = self.nodes.get(node)?;
+
+        match ndef.ntype {
+            NodeType::Receiver => self.receivers.get(ndef.inner.index).map(|r| r as &dyn Node),
+            NodeType::Processor => self
+                .processors
+                .get(ndef.inner.index)
+                .map(|p| p as &dyn Node),
+            NodeType::Exporter => self.exporters.get(ndef.inner.index).map(|e| e as &dyn Node),
+        }
+    }
+
+    /// Gets a mutable NodeWithPDataSender reference (processors and receivers).
+    #[must_use]
+    pub fn get_mut_sender(
```

Comment:
The name `get_mut_sender` initially confused me because it made me think of a channel sender. I think `get_mut_node_with_pdata_sender` will be (way too long I agree) but less ambiguous.